File: Second Seminar/SecondSeminar/src/main/java/com/example/SecondSeminar/post/presentation/PostControllerV2.java
package com.example.SecondSeminar.post.presentation;

import com.example.SecondSeminar.post.application.PostServiceV2;
import com.example.SecondSeminar.post.dto.request.PostCreateRequest;
import java.net.URI;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

@RequiredArgsConstructor
@RequestMapping("/api/v2/posts")
@RestController
public class PostControllerV2 {

    private static final String CUSTOM_AUTH_ID = "X-Auth-Id";
    private final PostServiceV2 postService;

    @PostMapping
    public ResponseEntity<Void> createPostV2(@RequestHeader(CUSTOM_AUTH_ID) Long memberId,
                                             @RequestPart MultipartFile image, PostCreateRequest request) {

        URI location = URI.create("/api/posts/v2" + postService.createV2(request, image, memberId));
        return ResponseEntity.created(location).build();
    }
}

File: Second Seminar/SecondSeminarKotlin/src/main/kotlin/com/example/SecondSeminarKotlin/post/presentation/PostControllerV2.kt
package com.example.SecondSeminarKotlin.post.presentation

import com.example.SecondSeminarKotlin.post.application.PostServiceV2
import com.example.SecondSeminarKotlin.post.dto.request.PostCreateRequest
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile
import java.net.URI

private const val CUSTOM_AUTH_ID = "X-Auth-Id"

@RestController
@RequestMapping("/api/v2/posts")
class PostControllerV2(
    private val postService: PostServiceV2
) {

    @PostMapping
    fun createPostV2(
        @RequestHeader(CUSTOM_AUTH_ID) memberId: Long,
        @RequestPart image: MultipartFile,
        request: PostCreateRequest
    ): ResponseEntity<Void> {
        val location = URI.create("/api/posts/v2/" + postService.createV2(request, image, memberId))
        return ResponseEntity.created(location).build()
    }

    @DeleteMapping("{postId}")
    fun deletePost(@PathVariable postId: Long): ResponseEntity<Void> {
        postService.deleteByIdV2(postId)
        return ResponseEntity.noContent().build()
    }
}